### ISSUE TITLE
Files with same name can produce same css classes

### ISSUE THREAD

**Issue by mlmorg** [css-modules/css-modulesify#15]
If you have two files with the same name (e.g. `foo.css`) both with the same local css class (e.g. `.bar`) but with different declarations, the resulting css looks something like:

```
._foo__bar {font-weight: bold;}
._foo__bar {font-weight: normal;}
```

What are your thoughts on the best way to handle this?

**Comment by rtsao** [comment]
The Webpack implementation allows you appends a hash of the content to the class name.
https://github.com/webpack/css-loader/blob/master/lib/getLocalIdent.js#L13
https://github.com/webpack/loader-utils/blob/master/index.js#L226

We need to be able to configure https://github.com/css-modules/postcss-modules-scope/blob/master/src/index.js#L159 to do something similar, but currently only the name and path are passed as arguments. Otherwise, we should be able to override `generateScopedName` through the options object.

**Comment by joshwnj** [comment]
Good call. I think it would be good to look into matching the behaviour of the webpack implementation but at this stage not sure how deep that goes - it may be a lot of work for not much value.  We may well find some other method of deduping that fits easier with the workflow.

**Comment by geelen** [comment]
This is strange, I thought the name being generated is the full *path* of the file, so it would be `._lib_foo__bar` and `._util_foo__bar`. That's how this line is intended to work: https://github.com/css-modules/postcss-modules-scope/blob/master/src/index.js#L160

**Comment by joshwnj** [comment]
@geelen yeah that's what I thought too... but it seems under certain circumstances it won't work that way. I'd like to try to get full paths working as well, because it gives some useful info to the developer. But I'm thinking it would be good to be able to use the hash for uniquifying as well (since that's what webpack is doing too).

**Comment by geelen** [comment]
If you're just hashing the path & the classname, won't the hashes be identical too?

**Comment by joshwnj** [comment]
nah I'm hashing the source css

**Comment by geelen** [comment]
Ah cool, alright i'll push through your change this arvo

**Comment by joshwnj** [comment]
cheers!

**Comment by geelen** [comment]
`postcss-modules-scope@0.0.8` released, guessing this is now covered under #16 so I'll close this

**Comment by mlmorg** [comment]
Not quite. `css-modules-loader-core` needs to have `postcss-modules-scope` bumped to `0.0.8`.

Better yet, bump everything to `1.0.0` so that you don't need to do those bumps manually anymore :)
